src/cpp/fastdds/xtypes/dynamic_types/idl_parser/IdlParser.hpp

```diff
@@ -0,0 +1,1725 @@
+            return parse(context.preprocess_string(idl_string), context);
+        }
+        else
+        {
+            return parse(idl_string, context);
+        }
+    }
+
+    //void get_all_types(
+    //    std::map<std::string, DynamicType::Ptr>& types_map)
+    //{
+    //    if (context_)
+    //    {
+    //        context_->module().fill_all_types(types_map);
+    //    }
+    //}
+
+    static std::string preprocess(
+            const std::string& idl_file,
+            const std::vector<std::string>& includes)
+    {
+        Context context;
+        context.include_paths = includes;
+        return context.preprocess_file(idl_file);
+    }
+
+private:
+
+    friend class Context;
+    //using LabelsCaseMemberPair = std::pair<std::vector<std::string>, Member>;
+
+    //peg::parser parser_;
+    Context* context_;
+
+    traits<DynamicType>::ref_type type_spec(
+            std::map<std::string, std::string>& state,
+            const std::string& type)
+    {
+        DynamicTypeBuilderFactory::_ref_type factory {DynamicTypeBuilderFactory::get_instance()};
+        DynamicTypeBuilder::_ref_type builder;
+        DynamicType::_ref_type xtype;
+
+        if (type == "boolean")
+        {
+            xtype = factory->get_primitive_type(TK_BOOLEAN);
+        }
+        else if (type == "int8")
+        {
+            xtype = factory->get_primitive_type(TK_INT8);
+        }
+        else if (type == "uint8")
+        {
+            xtype = factory->get_primitive_type(TK_UINT8);
+        }
+        else if (type == "int16")
+        {
+            xtype = factory->get_primitive_type(TK_INT16);
+        }
+        else if (type == "uint16")
+        {
+            xtype = factory->get_primitive_type(TK_UINT16);
+        }
+        else if (type == "int32")
+        {
+            xtype = factory->get_primitive_type(TK_INT32);
+        }
+        else if (type == "uint32")
+        {
+            xtype = factory->get_primitive_type(TK_UINT32);
+        }
+        else if (type == "int64")
+        {
+            xtype = factory->get_primitive_type(TK_INT64);
+        }
+        else if (type == "uint64")
+        {
+            xtype = factory->get_primitive_type(TK_UINT64);
+        }
+        else if (type == "float")
+        {
+            xtype = factory->get_primitive_type(TK_FLOAT32);
+        }
+        else if (type == "double")
+        {
+            xtype = factory->get_primitive_type(TK_FLOAT64);
+        }
+        else if (type == "long double")
+        {
+            xtype = factory->get_primitive_type(TK_FLOAT128);
+        }
+        else if (type == "char")
+        {
+            xtype = factory->get_primitive_type(TK_CHAR8);
+        }
+        else if (type == "wchar" || type == "char16")
+        {
+            xtype = factory->get_primitive_type(TK_CHAR16);
+        }
+        else if (type == "string")
+        {
+            TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
```

Comment:
You can use `DynamicTypeBuilderFactory::create_string_type`

Reply:
Done.